src/Raven.Server/Documents/Archive/ArchivalStorage.cs

```diff
@@ -0,0 +1,356 @@
+            CancellationToken cancellationToken)
+        {
+            var count = 0;
+            var currentTicks = options.CurrentTime.Ticks;
+
+            var archiveTree = options.Context.Transaction.InnerTransaction.ReadTree(DocumentsByArchive);
+            using (var it = archiveTree.Iterate(false))
+            {
+                if (it.Seek(Slices.BeforeAllKeys) == false)
+                {
+                    duration = null;
+                    return null;
+                }
+
+                var toArchive = new Dictionary<Slice, List<(Slice LowerId, string Id)>>();
+                duration = Stopwatch.StartNew();
+
+                do
+                {
+                    var entryTicks = it.CurrentKey.CreateReader().ReadBigEndianInt64();
+                    if (entryTicks > currentTicks)
+                        break;
+
+                    var ticksAsSlice = it.CurrentKey.Clone(options.Context.Transaction.InnerTransaction.Allocator);
+
+                    var docsToArchive = new List<(Slice LowerId, string Id)>();
+
+                    using (var multiIt = archiveTree.MultiRead(it.CurrentKey))
+                    {
+                        if (multiIt.Seek(Slices.BeforeAllKeys))
+                        {
+                            do
+                            {
+                                if (cancellationToken.IsCancellationRequested)
+                                    return toArchive;
+
+                                var clonedId = multiIt.CurrentKey.Clone(options.Context.Transaction.InnerTransaction.Allocator);
+
+                                using (var document = _database.DocumentsStorage.Get(options.Context, clonedId,
+                                           DocumentFields.Id | DocumentFields.Data | DocumentFields.ChangeVector))
+                                {
+                                    if (document is null || document.TryGetMetadata(out var metadata) == false || HasPassed(metadata, options.CurrentTime) == false)
+                                    {
+                                        docsToArchive.Add((clonedId, null));
+                                        continue;
+                                    }
+                                    docsToArchive.Add((clonedId, document.Id));
+                                }
+                            } while (multiIt.MoveNext() && docsToArchive.Count + count < options.AmountToTake);
+                        }
+                    }
+
+                    count += docsToArchive.Count;
+                    if (docsToArchive.Count>0)
+                        toArchive.Add(ticksAsSlice,docsToArchive);
+                    
+                    
+                } while (it.MoveNext() && count < options.AmountToTake);
+
+                return toArchive;
+            }
+        }
+
+        // private (bool AllExpired, string Id) GetConflictedArchival(DocumentsOperationContext context, DateTime currentTime, Slice clonedId)
```

Comment:
We should ignore conflicts for this, I think

Reply:
Ok, I'm going to clear those lines off, we'll be able to get back here if we missed something

Reply:
https://github.com/ravendb/ravendb/pull/15715/commits/a375dc625f5323f30906cc71c60e80867b0fea93